mvpa2/tests/test_group_clusterthr.py

```diff
@@ -0,0 +1,303 @@
+                           [1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0]])
+        expected_result = [5, 4, 3, 3, 2, 0, 2]  # 5 clusters of size 1,
+                                                 # 4 clusters of size 2 ...
+
+        test_ds = Dataset([test_M])
+        if i == 1:
+            test_M = test_M.astype(bool)
+
+        test_M_3d = np.hstack((test_M.flatten(),
+                               test_M.flatten())).reshape(2, 9, 16)
+        test_ds_3d = Dataset([test_M_3d])
+        # expected_result^2
+        expected_result_3d = np.array([0, 5, 0, 4, 0, 3, 0,
+                                       3, 0, 2, 0, 0, 0, 2])
+
+        size = 10000  # how many times bigger than test_M_3d
+        test_M_3d_big = np.hstack((test_M_3d.flatten(), np.zeros(144)))
+        test_M_3d_big = np.hstack((test_M_3d_big for i in range(size))
+                               ).reshape(3 * size, 9, 16)
+        test_ds_3d_big = Dataset([test_M_3d_big])
+        expected_result_3d_big = expected_result_3d * size
+
+        # check basic cluster size determination for plain arrays and datasets
+        # with a single sample
+        for t, e in ((test_M, expected_result),
+                  (test_ds, expected_result),
+                  (test_M_3d, expected_result_3d),
+                  (test_ds_3d, expected_result_3d),
+                  (test_M_3d_big, expected_result_3d_big),
+                  (test_ds_3d_big, expected_result_3d_big)):
+            assert_array_equal(np.bincount(gct._get_map_cluster_sizes(t))[1:],
+                               e)
+        # old
+        M = np.vstack([test_M_3d.flatten()]*10)
+        # new
+        ds = dataset_wizard([test_M_3d] * 10)
+        assert_array_equal(M, ds)
+        expected_result = Counter(np.hstack([gct._get_map_cluster_sizes(test_M_3d)]*10))
+        assert_array_equal(expected_result,
+                           gct.get_cluster_sizes(ds))
+
+        # test the same with some arbitrary per-feature threshold
+        thr = 4
+        labels, num = measurements.label(test_M_3d)
+        area = measurements.sum(test_M_3d, labels,
+                                index=np.arange(labels.max() + 1))
+        cluster_sizes_map = area[labels]  #.astype(int)
+        thresholded_cluster_sizes_map = cluster_sizes_map > thr
+        # old
+        M = np.vstack([cluster_sizes_map.flatten()]*10)
+        # new
+        ds = dataset_wizard([cluster_sizes_map] * 10)
+        assert_array_equal(M, ds)
+        expected_result = Counter(np.hstack([gct._get_map_cluster_sizes(
+                                         thresholded_cluster_sizes_map)]*10))
+        th_map = np.ones(cluster_sizes_map.flatten().shape) * thr
+        # threshold dataset by hand
+        ds.samples = ds.samples > th_map
+        assert_array_equal(expected_result,
+                           gct.get_cluster_sizes(ds))
+
+        #dumm_null_dist = range(10)
```

Comment:
what is up with all this commented out testing?  is it wrong, or ... ? should be prune or some comment at least left why commented out?